### ISSUE TITLE
Only the first AppRole on the roles list is used for authentication

### ISSUE THREAD

**Issue by jsynowiec** [ChannelMeter/vault-gatekeeper-mesos#80]
Please correct me if I'm wrong, but https://github.com/nemosupremo/vault-gatekeeper/blob/4faf5db35af0b3a0c02f4f0ed1d8d21fa421cb29/gatekeeper.go#L487 will always result in using the first item from the roles list, unless the requested role equals task name. Even if there are more AppRoles listed. The loop only checks if requested AppRole is allowed and doesn't overwrite `roleName` variable. Later it's the `roleName` not the `requestedRoleName` that's used in `GetRoleId` and `GetSecretId`.

Unit tests also only cover those two cases - single role or role and task name.

So now, if I specify my policy like so:
```json
"roles": [
  "service-sidecar-cache-reader",
  "service-sidecar-cache-writer"
]
```

and request a token specifying `role` to be `service-sidecar-cache-writer`, I'll get a Vault token for the `service-sidecar-cache-reader` AppRole.

**Comment by nemosupremo** [comment]
It looks like you are right - when we check if your requested role is valid, me must also update the role name.

**Comment by nemosupremo** [comment]
Fixed the behavior and updated the tests in v1.1.3
